src/main/java/org/kitteh/irc/client/library/event/user/UserNickChangeEvent.java

```diff
@@ -24,62 +24,67 @@
 package org.kitteh.irc.client.library.event.user;
 
 import org.kitteh.irc.client.library.Client;
-import org.kitteh.irc.client.library.element.Channel;
 import org.kitteh.irc.client.library.element.ServerMessage;
 import org.kitteh.irc.client.library.element.User;
 import org.kitteh.irc.client.library.event.abstractbase.ActorEventBase;
-import org.kitteh.irc.client.library.event.helper.ChannelUserListChange;
+import org.kitteh.irc.client.library.event.helper.Change;
+import org.kitteh.irc.client.library.event.helper.UserInfoChange;
 import org.kitteh.irc.client.library.util.Sanity;
 
 import javax.annotation.Nonnull;
 import java.util.List;
-import java.util.Optional;
 
 /**
  * A {@link User} has changed nickname!
  */
-public class UserNickChangeEvent extends ActorEventBase<User> implements ChannelUserListChange {
+public class UserNickChangeEvent extends ActorEventBase<User> implements UserInfoChange<String> {
     private final User newUser;
+    private final Change<String> change;
 
     /**
      * Creates the event.
      *
      * @param client client for which this is occurring
      * @param originalMessages original messages
-     * @param user user changing name
+     * @param oldUser oldUser changing name
      * @param newUser the new nickname
      */
-    public UserNickChangeEvent(@Nonnull Client client, @Nonnull List<ServerMessage> originalMessages, @Nonnull User user, @Nonnull User newUser) {
-        super(client, originalMessages, user);
+    public UserNickChangeEvent(@Nonnull Client client, @Nonnull List<ServerMessage> originalMessages, @Nonnull User oldUser, @Nonnull User newUser) {
+        super(client, originalMessages, oldUser);
         Sanity.nullCheck(newUser, "User cannot be null");
+        this.change = new Change<>(oldUser.getNick(), newUser.getNick());
         this.newUser = newUser;
     }
 
+    /**
+     * Gets the new user with the new nickname.
+     *
+     * @return the user with new nickname
+     */
     @Nonnull
     @Override
-    public Optional<Channel> getAffectedChannel() {
-        return Optional.empty();
+    public User getNewUser() {
+        return this.newUser;
     }
 
+    /**
+     * Convenience method
```

Comment:
"Gets the..." rather than "convenience method" also missing period at end of this.